deepchem/feat/molecule_featurizers/mol_graph_conv_featurizer.py

```diff
@@ -0,0 +1,196 @@
+
+
+def _construct_atom_feature(atom: RDKitAtom,
+                            h_bond_infos: List[Tuple[int, str]],
+                            sssr: List[Sequence]) -> List[float]:
+  """Construct an atom feature from a RDKit atom object.
+
+  Parameters
+  ----------
+  atom: rdkit.Chem.rdchem.Atom
+    RDKit atom object
+  h_bond_infos: List[Tuple[int, str]]
+    A list of tuple `(atom_index, hydrogen_bonding_type)`.
+    Basically, it is expected that this value is the return value of
+    `construct_hydrogen_bonding_info`. The `hydrogen_bonding_type`
+    value is "Acceptor" or "Donor".
+  sssr: List[Sequence]
+    The return value of `Chem.GetSymmSSSR(mol)`.
+    The value is a sequence of rings.
+
+  Returns
+  -------
+  List[float]
+    A one-hot vector of the atom feature.
+  """
+  atom_type = get_atom_type_one_hot(atom)
+  chirality = get_atom_chirality_one_hot(atom)
+  formal_charge = get_atom_formal_charge(atom)
+  partial_charge = get_atom_partial_charge(atom)
+  ring_size = get_atom_ring_size_one_hot(atom, sssr)
+  hybridization = get_atom_hybridization_one_hot(atom)
+  acceptor_donor = get_atom_hydrogen_bonding_one_hot(atom, h_bond_infos)
+  aromatic = get_atom_is_in_aromatic_one_hot(atom)
+  degree = get_atom_total_degree_one_hot(atom)
+  total_num = get_atom_total_num_Hs_one_hot(atom)
+  return atom_type + chirality + formal_charge + partial_charge + \
+    ring_size + hybridization + acceptor_donor + aromatic + degree + total_num
+
+
+def _construct_bond_feature(bond: RDKitBond) -> List[float]:
+  """Construct a bond feature from a RDKit bond object.
+
+  Parameters
+  ---------
+  bond: rdkit.Chem.rdchem.Bond
+    RDKit bond object
+
+  Returns
+  -------
+  List[float]
+    A one-hot vector of the bond feature.
+  """
+  bond_type = get_bond_type_one_hot(bond)
+  same_ring = get_bond_is_in_same_ring_one_hot(bond)
+  conjugated = get_bond_is_conjugated_one_hot(bond)
+  stereo = get_bond_stereo_one_hot(bond)
+  return bond_type + same_ring + conjugated + stereo
+
+
+class MolGraphConvFeaturizer(MolecularFeaturizer):
+  """This class is a featurizer of gerneral graph convolution networks for molecules.
+
+  The default node(atom) and edge(bond) representations are based on
+  `WeaveNet paper <https://arxiv.org/abs/1603.00856>`_. If you want to use your own representations,
+  you could use this class as a guide to define your original Featurizer. In many cases, it's enough
+  to modify return values of `construct_atom_feature` or `construct_bond_feature`.
+
+  The default node representation are constructed by concatenating the following values,
+  and the feature length is 38.
+
+  - Atom type: A one-hot vector of this atom, "C", "N", "O", "F", "P", "S", "Br", "I", "other atoms".
+  - Chirality: A one-hot vector of the chirality, "R" or "S".
+  - Formal charge: Integer electronic charge.
+  - Partial charge: Calculated partial charge.
+  - Ring sizes: A one-hot vector of the number of rings (3-8) that include this atom.
```

Comment:
I think you mean the size of the ring?  Not many atoms belong to three rings, much less eight!

Reply:
Yes! I fixed.